src/browser.js

```diff
@@ -10,10 +10,42 @@ exports.formatArgs = formatArgs;
 exports.save = save;
 exports.load = load;
 exports.useColors = useColors;
-exports.storage = 'undefined' != typeof chrome
-               && 'undefined' != typeof chrome.storage
-                  ? chrome.storage.local
-                  : localstorage();
+exports.storage = (function() {
+  var browser = window.browser || window.chrome;
+
+  if (typeof browser !== 'undefined' &&
+      typeof browser.storage !== 'undefined' &&
+      typeof browser.storage.local !== 'undefined') {
+    if (browser.storage.local.get.length === 2) {
+      // Chrome/Opera/Edge use the callback pattern
+      return browser.storage.local;
+    } else if (browser.storage.local.get.length === 1) {
```

Comment:
@esphen true that, I've seen the same (arity is 0 because they implement the functions with dynamic arguments).
The PR has been amended with a commit my team had in our private fork.